LoopKitUI/Views/ScheduleEditor.swift

```diff
@@ -351,7 +349,22 @@ struct ScheduleEditor<Value: Equatable, ValueContent: View, ValuePicker: View, A
         .disabled(tableDeletionState != .disabled || scheduleItems.count >= scheduleItemLimit)
     }
 
-    private func beginSaving() {
+    private func startSaving() {
+        guard mode == .settings || mode == .legacySettings else {
+            self.continueSaving()
+            return
+        }
+        
+        authenticate(authenticationChallengeDescription) {
+            switch $0 {
+            case .success: self.continueSaving()
+            case .failure(let error): self.presentedAlert = .saveError(error)
```

Comment:
This suggestion might sound odd, but I'm actually not sure we should present an alert when authentication fails here. Since an incorrect biometry reading keeps the user in the authentication context; generally they proceed only by 1) correctly authenticating or 2) cancelling. We can defer communication of auth errors to the auth modal itself.

Reply:
I don't know... I think this is a design call.  Normally I'd agree, but we're being ultra-conservative when it comes to communicating to the user what happened.  For example, if the user cancels the authentication, this will pop up an error:
![Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-03 at 15 07 30](https://user-images.githubusercontent.com/308993/89231877-165c0d00-d59b-11ea-8093-5e8c8e0dc2e9.png)

Pedantic?  Yes...but maybe not a bad idea?  I suggest we try and see how design review goes.  WDYT?